src/openforms/submissions/api/viewsets.py

```diff
@@ -246,6 +247,11 @@ def cosign(self, request, *args, **kwargs):
                 )
             )
 
+        if same_cosigner(request, submission):
+            raise PermissionDenied(
+                _("Submission cannot be co-signed by the initial submitter.")
+            )
+
```

Comment:
Discussed today, this validation can be moved to the `SearchSubmissionForCosignFormView` so that the validation takes place at an earlier point. 

Using an assert statement here will be sufficient so that it can be picked up during testing (but will be stripped in live environments).